src/main/java/net/glowstone/GlowWorld.java

```diff
@@ -1405,48 +1413,54 @@ public double getHumidity(int x, int z) {
      * Spawns an entity.
      *
      * @param location the {@link Location} to spawn the entity at
-     * @param clazz the class of the {@link Entity} to spawn
-     * @param reason the reason for the spawning of the entity
+     * @param clazz    the class of the {@link Entity} to spawn
+     * @param reason   the reason for the spawning of the entity
      * @return an instance of the spawned {@link Entity}
      * @throws IllegalArgumentException TODO: document the reason this can happen
      */
     public GlowEntity spawn(Location location, Class<? extends GlowEntity> clazz,
             SpawnReason reason) throws IllegalArgumentException {
-        GlowEntity entity = null;
 
         if (TNTPrimed.class.isAssignableFrom(clazz)) {
-            entity = new GlowTntPrimed(location, null);
+            return new GlowTntPrimed(location, null);
```

Comment:
This prevents the events from being called (under this block) for Primed TNT.

Reply:
Refactored in d78caf8ae29bfd9455a351a4f1dd267d2eb88b91 by overloading the GlowTntPrimed constructor so that it no longer needs separate logic.